### ISSUE TITLE
[steamshovel] Missing libraries stand-alone Mac app (Trac #1818)

### ISSUE THREAD

**Issue by icecube-trac** [icecube-trac/tix2#1810]
I have just tried using the `mac-steamshovel-app` target, and the resulting application bundle is non-functional when transferred to another machine. When launched the application fails due to a dynamic linking error; it cannot find any of the boost libraries it requires. Examining its contents this seems to be because they are not included as they should have been. 

```text
~/Desktop/Steamshovel.app/Contents/MacOS$ otool -L steamshovel
steamshovel:
	/usr/lib/libc++.1.dylib (compatibility version 1.0.0, current version 120.0.0)
	/usr/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 1197.1.1)
	@loader_path/../Frameworks/libicetray.dylib (compatibility version 0.0.0, current version 0.0.0)
	@loader_path/../Frameworks/libdataclasses.dylib (compatibility version 0.0.0, current version 0.0.0)
	@loader_path/../Frameworks/libdataio.dylib (compatibility version 0.0.0, current version 0.0.0)
	@loader_path/../Frameworks/libboost_python.dylib (compatibility version 0.0.0, current version 0.0.0)
	@loader_path/../Frameworks/libboost_system.dylib (compatibility version 0.0.0, current version 0.0.0)
	@loader_path/../Frameworks/libboost_signals.dylib (compatibility version 0.0.0, current version 0.0.0)
	@loader_path/../Frameworks/libboost_thread.dylib (compatibility version 0.0.0, current version 0.0.0)
	@loader_path/../Frameworks/libboost_date_time.dylib (compatibility version 0.0.0, current version 0.0.0)
	@loader_path/../Frameworks/libboost_serialization.dylib (compatibility version 0.0.0, current version 0.0.0)
	@loader_path/../Frameworks/libboost_filesystem.dylib (compatibility version 0.0.0, current version 0.0.0)
	@loader_path/../Frameworks/libboost_program_options.dylib (compatibility version 0.0.0, current version 0.0.0)
	@loader_path/../Frameworks/libboost_regex.dylib (compatibility version 0.0.0, current version 0.0.0)
	@loader_path/../Frameworks/libboost_iostreams.dylib (compatibility version 0.0.0, current version 0.0.0)
	/System/Library/Frameworks/Python.framework/Versions/2.7/Python (compatibility version 2.7.0, current version 2.7.5)
	/System/Library/Frameworks/AGL.framework/Versions/A/AGL (compatibility version 1.0.0, current version 1.0.0)
	/System/Library/Frameworks/OpenGL.framework/Versions/A/OpenGL (compatibility version 1.0.0, current version 1.0.0)
	@loader_path/../Frameworks/QtOpenGL.framework/Versions/4/QtOpenGL (compatibility version 4.8.0, current version 4.8.3)
	@loader_path/../Frameworks/QtGui.framework/Versions/4/QtGui (compatibility version 4.8.0, current version 4.8.3)
	@loader_path/../Frameworks/QtCore.framework/Versions/4/QtCore (compatibility version 4.8.0, current version 4.8.3)
	@loader_path/../Frameworks/libshovelart.dylib (compatibility version 0.0.0, current version 0.0.0)
	@loader_path/../Frameworks/libshovelio.dylib (compatibility version 0.0.0, current version 0.0.0)
	@loader_path/../Frameworks/QtSvg.framework/Versions/4/QtSvg (compatibility version 4.8.0, current version 4.8.3)
	@loader_path/../Frameworks/libserialization.dylib (compatibility version 0.0.0, current version 0.0.0)
~/Desktop/Steamshovel.app/Contents/MacOS$ ls ../Frameworks/
QtCore.framework	QtSvg.framework		libMinuit2.dylib	liberfa.1.dylib		libgsl.dylib		libhdf5_hl.7.dylib	libpal.0.dylib		libshovelart.dylib	usercustomize.py
QtGui.framework		icecube			libdataclasses.dylib	liberfa.dylib		libhdf5.7.dylib		libhdf5_hl.dylib	libpal.dylib		libshovelio.dylib
QtOpenGL.framework	libMinuit2.0.dylib	libdataio.dylib		libgsl.0.dylib		libhdf5.dylib		libicetray.dylib	libserialization.dylib	python
```

After discovering this I looked more carefully at the messages produced during the compilation and found this rather suspicious looking section:

```text
macdeployqt /Users/christopher/Work/IceCube/combo/build/Steamshovel.app -no-plugins -executable=/Users/christopher/Work/IceCube/combo/build/Steamshovel.app/Contents/MacOS/steamshovel-logviewer
ERROR: no file at "/usr/lib/libboost_python.dylib" 
ERROR: no file at "/usr/lib/libboost_system.dylib" 
ERROR: no file at "/usr/lib/libboost_signals.dylib" 
ERROR: no file at "/usr/lib/libboost_thread.dylib" 
ERROR: no file at "/usr/lib/libboost_date_time.dylib" 
ERROR: no file at "/usr/lib/libboost_serialization.dylib" 
ERROR: no file at "/usr/lib/libboost_filesystem.dylib" 
ERROR: no file at "/usr/lib/libboost_program_options.dylib" 
ERROR: no file at "/usr/lib/libboost_regex.dylib" 
ERROR: no file at "/usr/lib/libboost_iostreams.dylib"
```

It appears that the boost libraries are being searched for only in the non-standard location /usr/lib, while I have installed them in /usr/local/lib as is more normal. Clearly, they should be found in either location, but it isn't obvious to to me exactly where in macdeployapp.py this is supposed to happen; the above errors come from macdeployqt, but it seems odd that this tool, for which I can find little documentation, should be responsible for pulling in boost. 

Migrated from https://code.icecube.wisc.edu/ticket/1818
```json
{
    "status": "closed", 
    "changetime": "2019-02-13T14:13:18", 
    "description": "I have just tried using the `mac-steamshovel-app` target, and the resulting application bundle is non-functional when transferred to another machine. When launched the application fails due to a dynamic linking error; it cannot find any of the boost libraries it requires. Examining its contents this seems to be because they are not included as they should have been. \n\n{{{\n~/Desktop/Steamshovel.app/Contents/MacOS$ otool -L steamshovel\nsteamshovel:\n\t/usr/lib/libc++.1.dylib (compatibility version 1.0.0, current version 120.0.0)\n\t/usr/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 1197.1.1)\n\t@loader_path/../Frameworks/libicetray.dylib (compatibility version 0.0.0, current version 0.0.0)\n\t@loader_path/../Frameworks/libdataclasses.dylib (compatibility version 0.0.0, current version 0.0.0)\n\t@loader_path/../Frameworks/libdataio.dylib (compatibility version 0.0.0, current version 0.0.0)\n\t@loader_path/../Frameworks/libboost_python.dylib (compatibility version 0.0.0, current version 0.0.0)\n\t@loader_path/../Frameworks/libboost_system.dylib (compatibility version 0.0.0, current version 0.0.0)\n\t@loader_path/../Frameworks/libboost_signals.dylib (compatibility version 0.0.0, current version 0.0.0)\n\t@loader_path/../Frameworks/libboost_thread.dylib (compatibility version 0.0.0, current version 0.0.0)\n\t@loader_path/../Frameworks/libboost_date_time.dylib (compatibility version 0.0.0, current version 0.0.0)\n\t@loader_path/../Frameworks/libboost_serialization.dylib (compatibility version 0.0.0, current version 0.0.0)\n\t@loader_path/../Frameworks/libboost_filesystem.dylib (compatibility version 0.0.0, current version 0.0.0)\n\t@loader_path/../Frameworks/libboost_program_options.dylib (compatibility version 0.0.0, current version 0.0.0)\n\t@loader_path/../Frameworks/libboost_regex.dylib (compatibility version 0.0.0, current version 0.0.0)\n\t@loader_path/../Frameworks/libboost_iostreams.dylib (compatibility version 0.0.0, current version 0.0.0)\n\t/System/Library/Frameworks/Python.framework/Versions/2.7/Python (compatibility version 2.7.0, current version 2.7.5)\n\t/System/Library/Frameworks/AGL.framework/Versions/A/AGL (compatibility version 1.0.0, current version 1.0.0)\n\t/System/Library/Frameworks/OpenGL.framework/Versions/A/OpenGL (compatibility version 1.0.0, current version 1.0.0)\n\t@loader_path/../Frameworks/QtOpenGL.framework/Versions/4/QtOpenGL (compatibility version 4.8.0, current version 4.8.3)\n\t@loader_path/../Frameworks/QtGui.framework/Versions/4/QtGui (compatibility version 4.8.0, current version 4.8.3)\n\t@loader_path/../Frameworks/QtCore.framework/Versions/4/QtCore (compatibility version 4.8.0, current version 4.8.3)\n\t@loader_path/../Frameworks/libshovelart.dylib (compatibility version 0.0.0, current version 0.0.0)\n\t@loader_path/../Frameworks/libshovelio.dylib (compatibility version 0.0.0, current version 0.0.0)\n\t@loader_path/../Frameworks/QtSvg.framework/Versions/4/QtSvg (compatibility version 4.8.0, current version 4.8.3)\n\t@loader_path/../Frameworks/libserialization.dylib (compatibility version 0.0.0, current version 0.0.0)\n~/Desktop/Steamshovel.app/Contents/MacOS$ ls ../Frameworks/\nQtCore.framework\tQtSvg.framework\t\tlibMinuit2.dylib\tliberfa.1.dylib\t\tlibgsl.dylib\t\tlibhdf5_hl.7.dylib\tlibpal.0.dylib\t\tlibshovelart.dylib\tusercustomize.py\nQtGui.framework\t\ticecube\t\t\tlibdataclasses.dylib\tliberfa.dylib\t\tlibhdf5.7.dylib\t\tlibhdf5_hl.dylib\tlibpal.dylib\t\tlibshovelio.dylib\nQtOpenGL.framework\tlibMinuit2.0.dylib\tlibdataio.dylib\t\tlibgsl.0.dylib\t\tlibhdf5.dylib\t\tlibicetray.dylib\tlibserialization.dylib\tpython\n}}}\n\nAfter discovering this I looked more carefully at the messages produced during the compilation and found this rather suspicious looking section:\n{{{\nmacdeployqt /Users/christopher/Work/IceCube/combo/build/Steamshovel.app -no-plugins -executable=/Users/christopher/Work/IceCube/combo/build/Steamshovel.app/Contents/MacOS/steamshovel-logviewer\nERROR: no file at \"/usr/lib/libboost_python.dylib\" \nERROR: no file at \"/usr/lib/libboost_system.dylib\" \nERROR: no file at \"/usr/lib/libboost_signals.dylib\" \nERROR: no file at \"/usr/lib/libboost_thread.dylib\" \nERROR: no file at \"/usr/lib/libboost_date_time.dylib\" \nERROR: no file at \"/usr/lib/libboost_serialization.dylib\" \nERROR: no file at \"/usr/lib/libboost_filesystem.dylib\" \nERROR: no file at \"/usr/lib/libboost_program_options.dylib\" \nERROR: no file at \"/usr/lib/libboost_regex.dylib\" \nERROR: no file at \"/usr/lib/libboost_iostreams.dylib\"\n}}}\n\nIt appears that the boost libraries are being searched for only in the non-standard location /usr/lib, while I have installed them in /usr/local/lib as is more normal. Clearly, they should be found in either location, but it isn't obvious to to me exactly where in macdeployapp.py this is supposed to happen; the above errors come from macdeployqt, but it seems odd that this tool, for which I can find little documentation, should be responsible for pulling in boost. ", 
    "reporter": "cweaver", 
    "cc": "", 
    "resolution": "fixed", 
    "_ts": "1550067198646291", 
    "component": "combo core", 
    "summary": "[steamshovel] Missing libraries stand-alone Mac app", 
    "priority": "normal", 
    "keywords": "", 
    "time": "2016-08-12T22:20:49", 
    "milestone": "", 
    "owner": "sander.vanheule", 
    "type": "defect"
}
```

**Comment by icecube-trac** [comment]
Trac update at 20160812T22:39:35: cweaver commented:

> The missing boost libraries are indeed the problem; I realized that I could copy them into the bundle manually, and that hand-patched version does run on the other machine.

**Comment by icecube-trac** [comment]
Trac update at 20160815T14:46:00: 
* olivas changed milestone from "" to "September 2016"
* olivas changed status from "new" to "assigned"
* olivas changed owner from "" to "sander.vanheule"

**Comment by icecube-trac** [comment]
Trac update at 20160817T13:22:58: sander.vanheule changed summary from "Stand-alone Steamshovel target seems to be broken" to "[steamshovel] Missing libraries stand-alone Mac app"

**Comment by icecube-trac** [comment]
Trac update at 20160927T14:19:41: 
* cweaver changed resolution from "" to "fixed"
* cweaver changed status from "assigned" to "closed"
* cweaver commented:

> In changeset 150332:
> 
> ```text
> #!CommitTicketReference repository="IceCube" revision="150332"
> Try a little harder to include dependencies
> Fixes #1810
> ```

**Comment by icecube-trac** [comment]
Trac update at 20190213T14:13:18: 
* olivas changed milestone from "September 2016" to ""
* olivas commented:

> Milestone September 2016 deleted
